### DISCUSSION TITLE
Support Apache Arrow Format natively in the frontend

### DISCUSSION THREAD

**Discussion by mzy2240** [writer/writer-framework#30] [Ideas]
Pandas starts to support Arrow format from 2.0, and polars uses Arrow from beginning, which makes me think it would be nice to support Arrow in the frontend, to get rid of the hassle of converting dataframe to JSON and thus improve the overall performance. Here are the reference: https://arrow.apache.org/docs/js/

**Comment by mzy2240** [comment]
As another note, DuckDB has a good wasm [implementation](https://duckdb.org/docs/api/wasm/overview), so when it comes to sort, groupby type of operations for the data grid, it might be easier to perform those inside wasm and sync back to the backend (python) and frontend (js).

**Comment by ramedina86** [comment]
I definitely want to support Arrow. JSON serialisation for dataframes was good to get started but for bigger datasets it's almost a requirement.

Regarding your second point, DuckDB on WASM looks cool, I hadn't heard about it. I had a quick look and it's fast once it loads but startup times/bundle size are a bit prohibitive though. I was looking at Arquero which is lightweight and supports Arrow.

**Comment by ramedina86** [comment]
Arrow is now supported on 0.1.9.
